shared/shared/util/logging.py

```diff
@@ -0,0 +1,35 @@
+import logging
+import os
+import sys
+
+from shared.util import BadCodingError
+
+
+logger = logging.getLogger("datastore")
```

Comment:
F*** Github, this comment wasn't saved...

TLDR (it was quite long):

Try this: Instead of exporting `logger` export a function `getLogger(name: str) -> Logger`, which prepends `datastore.` to `name`, which is required. So you can get specific loggers in other modules with `logger = getLogger(__name__)`. This should allow for seeing, where a log statement in the logs came from and you can set verbosities for individual subloggers. It should be sufficient to attach the root logger (`datastore`) to the flask logger since all subloggers should use the configuration of the root logger if not specified differently.

Reply:
I explicitly did not do that since I find it quite annoying to init a logger in each module. Do you really think we need different verbosities for sub-loggers? I think one (or maybe 2, one for reader and one for writer) is enough...
Alternative: We subclass the `Logger` and make it print the module together with the message.

Reply:
> We subclass the Logger and make it print the module together with the message.

OK

the different levels might be good, but the more important thing is to see where the log message came from.

Reply:
Also, subclassing wasnt even neccessary, the formatter has an option for this ;) alternative to the path as is now would be just the module name. That's a little less verbose, but might be ambiguous in certain cases.